### ISSUE TITLE
predict_line_fluxes pairs the grid wavelength axis with the state's attenuated catalog — dusty Cue line fits are 23,000x low on the default approx=auto path

### ISSUE THREAD

**Issue by suchethac** [suchethac/tengri#1943]
`predict_line_fluxes` pairs the **grid's** wavelength axis with the **state's** attenuated luminosity array. On a dusty model with a discrete-catalog nebular backend, the line fluxes the fit objective consumes come out **~23,000x too small**, silently. This is the configuration the default `approx="auto"` policy produces.

## The defect

`SEDModel.predict_line_fluxes` (`src/tengri/forward/sed_model.py:~4895`) takes the fast branch whenever a grid exists, and sets both arrays from the grid:

```python
grid = getattr(self, "_nebular_grid_table", None)
if grid is not None:
    ...
    all_waves = jnp.asarray(grid.wavelengths)            # n = n_requested_targets (3)
    all_lums  = reconstruct_nebular_line_lums(nion, params, grid)
```

The shared redden tail (#1877, `:~4955`) may then replace `all_lums` — but not `all_waves`:

```python
if redden:
    _log_atten = state.derived.get("log_line_lums_attenuated") if state is not None else None
    if _log_atten is None:
        all_lums = self._attenuate_line_catalog(params, all_waves, all_lums)
    else:
        all_lums = pow10(jnp.asarray(_log_atten))        # n = full backend catalog (128)
```

`all_waves` is now the 3-entry grid axis while `all_lums` is the 128-entry backend catalog. The target match downstream indexes into `all_waves` and reads `all_lums` at those indices, so it returns the **first three entries of the full catalog** — the far-UV lines at 923–937 Å — under the labels Halpha / Hbeta / [OIII].

The `tolerance_aa` guard cannot catch it: the *wavelengths* match the requested targets exactly. Only the luminosities come from the wrong array.

## When it bites

All four must hold:

1. `FeaturePrecomp` is active (so `_nebular_grid_table` exists),
2. a `state` is passed — which `loss_functions._build_prediction` does, because `_fast_line_measurement` is `False` while `approx.feature_precomp` is `True`,
3. the state publishes `log_line_lums_attenuated` — i.e. a dust component, and
4. `use_grid` is `False`, so the nebular component ran its exact branch and published the catalog. **Dust is what forces this** (`must_materialize_sed`, #1281/#1748).

So: **dusty + a discrete-catalog backend (Cue/CloudyGrid) + a line-flux channel.** Dust-free models are unaffected — there `use_grid` is `True`, no attenuated catalog is published, and the fallback screen runs against the grid's own axis, which is correctly paired.

## Measured

`fsps_prsc_miles_chabrier` SSP, `sfh=dpl` (5 free), `dust=two_component/calzetti`, `neb=cue`, z=0.1, 3 SDSS bands + 3 line fluxes at 5%.

Line fluxes at the same params, same model, only `state=` differing:

| call | Halpha [erg/s/cm2] |
|---|---|
| exact model (`approx=None`) | `9.150758e-12` |
| LUT model, `state=None` | `9.150752e-12` |
| **LUT model, `state` supplied** | **`3.962955e-16`** |

Constant across the prior volume — a fixed factor, not interpolation error:

| `log_total_mass` | exact Halpha | LUT Halpha | deviation |
|---|---|---|---|
| 12.23 | 9.105777e-12 | 3.943475e-16 | 99.9957 % |
| 11.00 | 5.361885e-13 | 2.322092e-17 | 99.9957 % |
| 9.00  | 5.361885e-15 | 2.322092e-19 | 99.9957 % |

The factor closes exactly against the catalog mismatch: `10**atten[state_idx] / 10**atten[grid_idx]` = `2.479454e+44 / 1.073787e+40` = **23,090**, against a measured flux ratio of **23,093**.

Effect on the fit objective (same point, identical `data_args`, verified by cross-evaluation):

| | exact | with LUT |
|---|---|---|
| objective | 1197.617745 | 1198.755586 |
| `d(nlp)/d(sfh_dpl_age_gyr)` | 3.937704e+00 | 1.680182e+00 |
| `d(nlp)/d(sfh_dpl_log_total_mass)` | -1.206182e+01 | -6.339191e+00 |
| `d(nlp)/d(sfh_dpl_beta)` | -4.617320e-01 | -9.044962e-02 |

Gradient components are **47–80 % off** — the line channel is pinned near zero, so it contributes almost no derivative. The objective offset scales as 1/sigma^2 (1.138e+00 at 5 % errors, 1.138e+02 at 0.5 %), the signature of a constant forward bias entering the gradient multiplied by SNR (#1671).

Attribution control in the same run: `WavePrecomp` alone is clean (max relative gradient deviation **4.6e-07**); `FeaturePrecomp` alone reproduces the whole effect. So this is not the photometry LUT.

## Suggested fix

The two arrays must come from one catalog. When the redden tail takes the state's published `log_line_lums_attenuated`, it must also take the state's `line_waves`; the grid axis is only correct for the grid's own reconstruction. Keeping #1877's canonical single screen and #1477's stateless fast path both work — they just must not be mixed.

## Labels

`area:nebular` `area:forward` `area:inference` `bug` `silent-failure`
